int32 allied_type
---
float32 gain
int32 error